.circleci/config.yml

```diff
@@ -272,6 +276,71 @@ jobs:
             docker login --username "${DOCKER_USER}" --password "${DOCKER_PASSWORD}"
             ./gradlew --no-daemon --parallel "-Pbranch=${CIRCLE_BRANCH}" dockerUpload
 
+  extractAPISpec:
+    executor: medium_executor
+    steps:
+      - prepare
+      - attach_workspace:
+          at: ~/project
+      - run:
+          name: Build runnable Teku binaries
+          command: |
+            ./gradlew --no-daemon --parallel installDist
+      - run:
+          name: Extract OpenAPI JSON
+          command: |
+            build/install/teku/bin/teku --network=mainnet --Xinterop-enabled=true --rest-api-docs-enabled=true &
+            TEKU_PID=$!
+            wget -O openapi.json http://localhost:5051/swagger-docs
+            kill $TEKU_PID
+      - store_artifacts:
+          path: openapi.json
+      - persist_to_workspace:
+          root: ~/project
+          paths:
+            - ./
+
+  publishAPIDoc:
+    executor: node_executor
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ~/project
+      - add_ssh_keys:
+          fingerprints:
+            - '45:36:58:67:43:15:c6:5f:2c:58:ec:7f:71:e2:e6:ef'
+      - run:
+          name: Set Git user params
+          command: |
+            git config --global user.name $CIRCLE_USERNAME
+            git config --global user.email "${CIRCLE_USERNAME}@users.noreply.github.com"
+      - restore_cache:
+          keys:
+            - deps-{{ checksum "./docs/package.json" }}
+      - run:
+          name: Install dependencies
+          working_directory: ./docs
+          command: |
+            npm install
+      - save_cache:
+          paths:
+            - ./docs/node_modules
+          key: deps-{{ checksum "./docs/package.json" }}
+      - run:
+          name: Generate API doc site files
+          working_directory: ./docs
+          command: |
+            npm run generate-versions -- --spec=../openapi.json --tag=${CIRCLE_TAG}
+      - deploy:
+          name: Commit and push doc changes
+          command: |
+            git add docs
+            git commit -m "Automated update of JSON specs and version files" \
+                       -m "[skip ci] Circle CI build ${CIRCLE_BUILD_NUM}, see ${CIRCLE_BUILD_URL}" \
+                       -m "Branch: ${CIRCLE_BRANCH}, commit: ${CIRCLE_SHA1}, tag:${CIRCLE_TAG-none}"
```

Comment:
I'm not familiar with this syntax - does `${CIRCLE_TAG-none}` mean "use the contents of `CIRCLE_TAG`, or if it's empty, use `none`? If so, cool. If not, you'll likely want to create a new env var for this above like so

`[ -n "$CIRCLE_TAG" ] && TAG_NAME="$CIRCLE_TAG" || TAG_NAME="none"`

though this is probably better expressed as an if/else rather than a one-liner - it's more readable that way.

Reply:
I've always used `${CIRCLE_TAG:-none}` to mean use `none` unless `CIRCLE_TAG` is set.  Seems to work without the `:` too though.

Reply:
> ${parameter-default} and ${parameter:-default} are almost equivalent. The extra : makes a difference only when parameter has been declared, but is null.

https://www.tldp.org/LDP/abs/html/parameter-substitution.html